src/ScalarMessaging.ts

```diff
@@ -322,6 +346,35 @@ function inviteUser(event: MessageEvent<any>, roomId: string, userId: string): v
     });
 }
 
+function kickUser(event: MessageEvent<any>, roomId: string, userId: string): void {
+    logger.log(`Received request to kick ${userId} from room ${roomId}`);
+    const client = MatrixClientPeg.get();
+    if (!client) {
+        sendError(event, _t('You need to be logged in.'));
+        return;
+    }
+    const room = client.getRoom(roomId);
+    if (room) {
+        // if they are already not in the room we can resolve immediately.
+        const member = room.getMember(userId);
+        if (!member || (member && member.membership === "leave")) {
```

Comment:
```suggestion
        if (!member || getEffectiveMembership(member.membership) === EffectiveMembership.Leave) {
```

to cover bans and such - `member` also shouldn't be falsy in the second branch.

Reply:
(this stuff will need importing)